testsuite/scripts/src/test/java/org/wildfly/scripts/test/StandaloneScriptTestCase.java

```diff
@@ -86,6 +89,28 @@ public static Collection<Object> data() {
         return result;
     }
 
+    @Test
+    public void testBatchScriptJavaOptsToEscape() throws Exception {
+        Assume.assumeTrue(!TestSuiteEnvironment.isWindows() && Shell.BASH.isSupported());
+        final String variableToEscape = "-Dhttp.nonProxyHosts=localhost|127.0.0.1|10.10.10.*";
+        ServerConfigurator.appendJavaOpts(ServerHelper.JBOSS_HOME, "standalone", variableToEscape);
+        try (ScriptProcess script = new ScriptProcess(ServerHelper.JBOSS_HOME, STANDALONE_BASE_NAME, Shell.BATCH, ServerHelper.TIMEOUT)) {
+            testScript(script);
+        }
+        ServerConfigurator.removeJavaOpts(ServerHelper.JBOSS_HOME, "standalone", variableToEscape);
+    }
+
+    @Test
+    public void testBatchScriptJavaOptsEscaped() throws Exception {
+        Assume.assumeTrue(!TestSuiteEnvironment.isWindows() && Shell.BASH.isSupported());
```

Comment:
The assume condition needs to be fixed: `Assume.assumeTrue(TestSuiteEnvironment.isWindows())`